src/unistd.rs

```diff
@@ -819,6 +819,64 @@ pub fn mkstemp<P: ?Sized + NixPath>(template: &P) -> Result<(RawFd, PathBuf)> {
     Ok((fd, PathBuf::from(pathname)))
 }
 
+libc_bitflags!{
+    pub flags AccessFlags: c_int {
+        R_OK,
+        W_OK,
+        X_OK,
+    }
+}
+
+pub enum AccessMode {
+    F_OK,
+    Flags(AccessFlags),
+}
+
+impl AccessMode {
+    fn to_libc_type(&self) -> c_int {
+        match self {
+            &AccessMode::F_OK => libc::F_OK,
+            &AccessMode::Flags(flags) => flags.bits(),
+        }
+    }
+}
+
+/// Check whether a file exists or whether the current process can access a file.
+///
+/// # Example
+/// ```
+/// use nix::unistd::*;
+///
+/// let devnull_exists = access("/dev/null", AccessMode::F_OK);
+/// assert!(devnull_exists.is_ok());
+/// ```
+pub fn access<P: ?Sized + NixPath>(path: &P, mode: AccessMode) -> Result<()> {
+    let res = try!(path.with_nix_path(|path|
+        unsafe {
+            libc::access(path.as_ptr(), mode.to_libc_type())
+        }
+    ));
+    Errno::result(res).map(drop)
+}
+
+/// Check whether a file exists or whether the current process can access a file,
```

Comment:
Please link to the man page for this. See the `fork()` function doccomments for an example.